classes/Image.php

```diff
@@ -635,11 +638,16 @@ public function deleteImage($forceDelete = false)
 
         // Delete auto-generated images
         $image_types = ImageType::getImagesTypes();
+        $sfContainer = SymfonyContainer::getInstance();
```

Comment:
Can't you store the instance of the symfony container in a property inside the class ? Because you're retrieving it n times in a loop, it would improve the performance if you were doing it only once.

Reply:
oh nice catch 👍

Reply:
So, at first I applied your feedback, and set `$this->symfonyContainer = SymfonyContainer::getInstance()` in the constructor, but I realised it might not be a good idea, since we would call the service container even if we don't need it.

As a matter of performance, `SymfonyContainer::getInstance()` is a singleton, once the container has been instantiated, we don't instantiate it anymore, it's no more costly than let's say calling a property from another class.

Reply:
I think there is no need for this. As improvement is negligible here (IO versus nanoseconds gain). I think size does not matter (here :trollface: )